server/services/location.js

```diff
@@ -0,0 +1,65 @@
+import axios from 'axios';
+import GOOGLE_API_KEY from '../config/development';
+
+const SEARCH_RADIUS_IN_METERS = 500;
+
+module.exports = class LocationService {
+    constructor() {}
+
+    async isNearAStore() {
+        const userLocation = await this.getUserLocation();
+        if (!userLocation){
+            return false;
+        }
+        const nearbyStores = await this.findNearbyStores(userLocation.lat, userLocation.lng);
+        return nearbyStores && nearbyStores.length > 0;
+    }
+
+    async getUserLocation() {
+        try {
+            const response = await axios.post('https://www.googleapis.com/geolocation/v1/geolocate?key=' + GOOGLE_API_KEY);
```

Comment:
Should this request have a body or something? It isn't clear how the location of the user could be inferred using just a generic URL without any parameters or anything. I think it might be working right now because the server making the request is located in the same place as the user?

Reply:
ugh

ugh

UGH

Man, I don't know what I was thinking. Yeah that is of course what is happening, and this can't be on the server unless we pass device data in from the client. But also getting an accurate location for mobile devices requires getting a bunch of info from the Android and iOS SDKs, which we are not going to have time to do.

So I'm left wondering if geofencing like this is really a viable approach for the purposes of this hackathon, or if we should just let the shopper enter an address or a zip code and show lists from that nearby area.

I'm thinking it's probably the latter which makes this PR moot. :(  What do you think @JimmyPruitt?

Reply:
Asking for a zip or address seems reasonable to me. I wouldn't say that makes this moot though. We could just modify this service a little bit to assume the shopper's rough whereabouts are known, but keep the lookup for stores nearby, 'cause that's still handy.

Reply:
cool, that works for me